src/components/data_submission/ds_study_information.js

```diff
@@ -0,0 +1,147 @@
+import { useState, useEffect } from 'react';
+import { h, h2, div } from 'react-hyperscript-helpers';
+import { Notifications, isEmailAddress } from '../../libs/utils';
+import { User } from '../../libs/ajax';
+import { FormFieldTypes, FormField, FormTable } from '../forms/forms';
+
+export default function DataSubmissionStudyInformation(props) {
+  const { onChange } = props;
+  const [user, setUser] = useState();
+
+  const updateUserAndFields = async () => {
+    const me = await User.getMe();
+    setUser(me);
+    onChange({key: 'dataSubmitterId', value: me.dacUserId});
+  };
+
+  //init hook, need to make ajax calls here
+  useEffect(() => {
+    const init = async () => {
+      try {
+        updateUserAndFields({ setUser, ...props });
+      } catch (error) {
+        Notifications.showError({text: 'Error: Unable to retrieve user data from server'});
+      }
+    };
+
+    init();
+  // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, []); // need the eslint to force useEffect to only run on initialization
```

Comment:
Ignoring the exhaustive-deps rule is frowned upon. Here's a thread on Github with gaeron (Dan Abramov, software engineer on the React team) explaining his concerns
https://github.com/facebook/create-react-app/issues/6880

Basically turning the rule off causes developers to miss out on code smells or logical issues with the method in the useEffect itself. In this case, we see that assigning the required dependencies results in an infinite re-render loop.

So what's causing the issue? The two dependencies on this useEffect are `onChange` and `updateUserAndFields`. `updateUserAndFields` is defined in this component, so the only way that can trigger this function to re-fire is if the props are updated. And the only thing on props right now is `onChange`. So somehow, onChange is being updated when this useEffect is fired, causing the component to re-render and fire the useEffect again, resulting in the infinite loop.

Okay, so let's look at the parent component `DataSubmissionForm` and see what's up there. onChange looks to be an anonymous function that is defined directly on the component as a prop. This means that if `DataSubmissionForm` re-renders, the `onChange` function is re-defined, meaning the props update, triggering the re-render.

That said, what's causing the parent component to re-render? Well the onChange is updating `formData` with a cloned value. A cloned value means a new object reference. React triggers re-renders if props or state variables update their values. If the variable is an object, then React will compare object references (not the contents) to see if they are the same value. 

In this instance a cloned copy is a new object which means a different reference. This will trigger a re-render on the parent and start the infinite loop you're seeing. To avoid the complete page re-render, update the target keys on `formData`.

I know React promotes immutability, but it does so with regards to `this.state` itself (meaning you shouldn't directly update state unless using a hook). Updating `formData` in this context is fine, especially considering that each sub-component is going to rely on formData as a source of truth for data and rendering triggers.

For further reference, here's another post explaining the above in greater detail
https://blog.logrocket.com/immutability-in-react-ebe55253a1cc/

Reply:
Thanks for the explanation! Updated the component to not have an empty deps array, and cleaned up the sideeffects fallout!